### DISCUSSION TITLE
One user can clone, the other cannot. Why?

### DISCUSSION THREAD

**Discussion by LukasvdWiel** [community/community#23286] [New to GitHub]
<p>Hello,</p>
<p>we have a small development team, and about half a year ago we introduced version management by means of git.</p>
<p>I set up a bare remote repository, and each of the developers has his own local repository.</p>
<p>I put in the code and everything worked well for a while.</p>
<p>However, when I want to make a fresh clone, I get the warning:</p>
<pre><code>warning: You appear to have cloned an empty repository.
</code></pre>
<p>While my colleague gets all the directories and files of the project with the same command. And if I request the log, I get</p>
<pre><code>fatal: Not a git repository
</code></pre>
<p>Also, when I wish to see what has happened in the remote repository, I get</p>
<pre><code>(11:19 lukas@login01 remoteRepository) &gt; git log
fatal: bad default revision 'HEAD'
</code></pre>
<p>When my colleague does the same, he gets all the messages of the various commits.</p>
<p>We are both in the same user group that has read permissions to the remote repository.</p>
<p>I was able to interact properly with it in the past. I have also made commits to the repo.</p>
<p>However, I must have inadvertently messed something up that disables me to interact properly with the remote repository now but I have no idea what it could be. There is nothing about GIT in either of our environments that could indicate a difference…</p>
<p>Any ideas would be very welcome!</p>

**Comment by majormoses** [comment]
<p>Hard to say without more information, can you please post the output of <code>git remote -v</code> on both setups? Typically this means that your local repository is not tracking the upstream repository. Here is an example where you can see that my “fork” of the repository is where I do all my work, I create pull requests “upstream”, and only allow pulling from the upstream.</p>
<pre><code>$ git remote -v
origin git@github.com:majormoses/aws.git (fetch)
origin git@github.com:majormoses/aws.git (push)
upstream git@github.com:chef-cookbooks/aws.git (fetch)
upstream no_push (push)
</code></pre>
<p>In this example <a href="https://github.com/sensu-plugins/community/blob/master/templates/CONTRIBUTING.md#workflow" rel="nofollow noopener">https://github.com/sensu-plugins/community/blob/master/templates/CONTRIBUTING.md#workflow</a> we outline how users who do not have direct write access to repositories can contribute, sync their changes from upstream, and create a pull request to get the changes merged back in. While in many cases adding the extra steps to avoid pushing to the upstream is not neccesary (due to permission issues) you will run into projects that still expect you to use your fork even if you have write access.</p>
<p>I will be honest I have not managed a git server in over a decade, I am <em>generally</em> pretty happy with github and I am fine letting them managing those aspects of permissions and such. For more granular control of permissions I recalled using <a href="https://gitolite.com/gitolite/index.html" rel="nofollow noopener">https://gitolite.com/gitolite/index.html</a>.</p>

**Comment by LukasvdWiel** [comment]
<p>Hi Majormoses,</p>
<p>thank you for taking the time to help me out.</p>
<p>My colleague where everything works fine gets, in his own local repo:</p>
<pre><code>origin /quanta1/home/gtecton/remoteRepository (fetch)
origin /quanta1/home/gtecton/remoteRepository (push)
</code></pre>
<p>And I get exactly the same in my own:</p>
<pre><code>origin /quanta1/home/gtecton/remoteRepository (fetch)
origin /quanta1/home/gtecton/remoteRepository (push)
</code></pre>
<p>But when I <em>git pull</em> in my own local:</p>
<pre><code>Your configuration specifies to merge with the ref 'master'
from the remote, but no such ref was fetched.
</code></pre>
<p>When my colleague pulls just fine. However, when we <em>git branch</em>, we both get</p>
<pre><code>*master
</code></pre>
<p>plus he gets two additional branches he has made, while I have only the master.</p>
<p>Does that help?</p>
<p>P.S. The links were good reading material, for which my thanks, but they did not really provide an answer. The workflow was OK the way it was, and it all worked in the past.</p>

**Comment by majormoses** [comment]
<p>Perhaps trying to fetch the origin will help?</p>
<pre><code>git fetch origin master
</code></pre>
<p>If that does not help trying adding a -v option and post the output here.</p>

**Comment by LukasvdWiel** [comment]
<p>Both verbose and non verbose give me the same:</p>
<pre><code>fatal: Couldn't find remote ref master
fatal: The remote end hung up unexpectedly
</code></pre>
<p>Our <em>git config --list</em> contain differences, though…</p>
<p>My colleague has</p>
<pre><code>core.repositoryformatversion=0
core.filemode=true
core.bare=false
core.logallrefupdates=true
remote.origin.url=/quanta1/home/gtecton/remoteRepository
remote.origin.fetch=+refs/heads/*:refs/remotes/origin/*
</code></pre>
<p>I have exactly that, but I have an additional:</p>
<pre><code>branch.master.remote=origin
branch.master.merge=refs/heads/master
</code></pre>
<p>Edit: If I look at it, I think it might be a permission error after all…</p>
<pre><code>(11:52 lukas@login01 newLocal) &gt; git push origin master
Counting objects: 2677, done.
Delta compression using up to 48 threads.
Compressing objects: 100% (2251/2251), done.
Writing objects: 100% (2677/2677), 31.39 MiB | 2.61 MiB/s, done.
Total 2677 (delta 914), reused 1447 (delta 344)
error: unable to resolve reference refs/heads/master: Permission denied
remote: error: failed to lock refs/heads/master
To /quanta1/home/gtecton/remoteRepository
 ! [remote rejected] master -&gt; master (failed to lock)
error: failed to push some refs to '/quanta1/home/gtecton/remoteRepository'
</code></pre>
<p>So I will investigate that for the time being…</p>
<p>Second edit. I explicitly set checkout to master.</p>
<p>Now I can git log and get the list of commit messages. Much improve.</p>
<p><em>Git pull</em> still gives me</p>
<pre><code>Your configuration specifies to merge with the ref 'master'
from the remote, but no such ref was fetched.
</code></pre>
<p>and <em>git ls-remote origin -v</em> gives me an empty list, while a local <em>git remote</em> still returns</p>
<pre><code>origin	/quanta1/home/gtecton/remoteRepository (fetch)
origin	/quanta1/home/gtecton/remoteRepository (push)
</code></pre>
<p>so it seems the remote repository was somehow affected… Or is that a wrong conclusion?</p>

**Comment by LukasvdWiel** [comment, marked as answer]
<p>Solution was found.</p>
<p>All user but one (let us name him John) belong to a specific group.</p>
<p>All users but me also belong to a specific other group.</p>
<p>John made his very first commit, and his primary user group was the group of which all other users also were member, but I was not. All other users had the same group policy could access the files normally. However, I was the odd one out.</p>
<p>For some reason the permission error was drowned somewhere down the river.</p>
<p>I added myself to the group, and now I can operate everything as usual.</p>
<p>Best wishes!</p>

**Comment by danihall07** [comment]
<p>What does that mean</p>

**Comment by CS50FanBoy** [comment]
<p>didnt understand none what you were saying</p>

**Comment by Ladyking0612** [comment]
Clone anwser
